sdk/src/main/scala/com/horizen/AbstractWallet.scala

```diff
@@ -114,9 +114,8 @@ abstract class AbstractWallet[
       case Some(nonce) => nonce
       case None => allSecrets.count(_.isInstanceOf[T])
     }
-    val salt: Array[Byte] = secretCreator.salt()
     for (_ <- 0 to secretStorageSize) {
-      val seed = Blake2b256.hash(Bytes.concat(this.seed, Ints.toByteArray(nonce), salt))
+      val seed = Blake2b256.hash(Bytes.concat(this.seed, Ints.toByteArray(nonce), secretCreator.salt()))
```

Comment:
`val salt: Array[Byte] = secretCreator.salt()` can be called once at the beginning of the method

Reply:
Ok

Reply:
Actually this field is static in secret creator, but if you want to initialize a variable explicitly - I did it